src/test/java/org/springframework/data/elasticsearch/core/ElasticsearchTemplateTests.java

```diff
@@ -1280,6 +1280,32 @@ public void shouldReturnResultsWithStreamForGivenCriteriaQuery() {
 		assertThat(sampleEntities).hasSize(30);
 	}
 
+	@Test // DATAES-764
+	public void shouldNotFailOnManySearchForStreams() {
+
+		// given
+		List<IndexQuery> entities = createSampleEntitiesWithMessage("Test message", 30);
+		operations.bulkIndex(entities, index);
+		indexOperations.refresh();
+		CriteriaQuery criteriaQuery = new CriteriaQuery(new Criteria());
+		criteriaQuery.setPageable(PageRequest.of(0, 10));
+
+		// when
+		Throwable throwable = catchThrowable(() -> {
+			for (int i = 0; i < 1000; i++) {
+				try (CloseableIterator<SearchHit<SampleEntity>> stream = operations.searchForStream(criteriaQuery,
+						SampleEntity.class, index)) {
+					while (stream.hasNext()) {
+						stream.next();
+					}
+				}
+			}
+		});
+
+		// then
+		assertThat(throwable).isNull();
+	}
+
```

Comment:
Please don't test if Elasticsearch produces an error. The test should make sure that when the user calls the `close()` method of the `CloseableIterator` this in return calls the `clearScroll` callback (what this callback sends to Elasticsearch is out of the scope of the `StreamQueries` class).

This is a `StreamQueriesTest` class that does this (fails on the existing code and succeeds after the changes):

```
/*
 * Copyright 2020 the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      https://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.springframework.data.elasticsearch.core;

import static org.assertj.core.api.Assertions.*;

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;
import java.util.concurrent.atomic.AtomicBoolean;

import org.junit.jupiter.api.Test;
import org.springframework.data.domain.PageImpl;
import org.springframework.data.util.CloseableIterator;
import org.springframework.lang.Nullable;

class StreamQueriesTest {

	@Test // DATAES-764
	void closingTheIteratorShouldCallClearScroll() {
		List<String> results = new ArrayList<>();
		results.add("one");

		ScrolledPage<String> page = new ScrolledPageImpl("42", results);

		AtomicBoolean clearScrollCalled = new AtomicBoolean(false);

		CloseableIterator<String> closeableIterator = StreamQueries.streamResults(page, //
				scrollId -> new ScrolledPageImpl(scrollId, Collections.emptyList()), //
				scrollId -> clearScrollCalled.set(true));

		while (closeableIterator.hasNext()) {
			closeableIterator.next();
		}
		closeableIterator.close();

		assertThat(clearScrollCalled).isTrue();
	}

	static class ScrolledPageImpl extends PageImpl<String> implements ScrolledPage<String> {

		@Nullable private String scrollId;

		public ScrolledPageImpl(String scrollId, List<String> content) {
			super(content);
			this.scrollId = scrollId;
		}

		@Override
		@Nullable
		public String getScrollId() {
			return scrollId;
		}
	}
}

```

Reply:
Thanks, I'll add it.